File: android/src/main/java/com/audiotoggle/plugin/AudioTogglePlugin.java
package com.audiotoggle.plugin;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "AudioToggle")
public class AudioTogglePlugin extends Plugin {

    private AudioToggle implementation = new AudioToggle();

    @PluginMethod
    public void setAudioMode(PluginCall call) {
        String mode = call.getString("mode");
        if (mode == null) {
            call.reject("mode is required");
            return;
        }

        boolean success = implementation.setAudioMode(mode, getContext());
        if (success) {
            call.resolve();
        } else {
            call.reject("Failed to set audio mode");
        }
    }
}

File: ios/Sources/AudioTogglePlugin/AudioTogglePlugin.swift
import Foundation
import Capacitor
import AVFoundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AudioTogglePlugin)
public class AudioTogglePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "AudioTogglePlugin"
    public let jsName = "AudioToggle"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "setAudioMode", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = AudioToggle()

    @objc func setAudioMode(_ call: CAPPluginCall) {
        guard let mode = call.getString("mode") else {
            call.reject("mode is required")
            return
        }

        do {
            try implementation.setAudioMode(mode: mode)
            call.resolve()
        } catch let error {
            call.reject("Failed to set audio mode: \(error.localizedDescription)")
        }
    }
}
